hibernate-core/src/test/java/org/hibernate/test/cut/CompositeUserTypeTest.java

```diff
@@ -67,6 +83,21 @@ public void testCompositeUserType() {
 		t.commit();
 		s.close();
 	}
+
+	@Test
+	public void testMetamodel() {
+		MetamodelImplementor metamodel = sessionFactory().getMetamodel();
+		PersistentAttributeDescriptor<? super Transaction, ?> value = metamodel.managedType(Transaction.class).getAttribute("value");
+		assertEquals(value.getPersistentAttributeType(), Attribute.PersistentAttributeType.EMBEDDED);
```

Comment:
It seems we might need to swap the two parameter orders for `assertEquals()`. The issue only manifests itself when testing is broken, however.

Reply:
After fixing this, I see nothing else blocking this PR. Any concerns @Sanne ?